packages/driver/src/cy/commands/actions/type.ts

```diff
@@ -235,7 +237,8 @@ export default function (Commands, Cypress, cy, state, config) {
         // when we send {Enter} KeyboardEvent to the input fields.
         // Because of that, we don't have to click the submit buttons.
         // Otherwise, we trigger submit events twice.
-        if (!Cypress.isBrowser('firefox')) {
+        // But after Firefox 98, submit event isn't automatically fired.
+        if (!isFirefoxBefore98 || !Cypress.isBrowser('firefox')) {
```

Comment:
The second part of this conditional is redundant. Three cases:
FF 98: (!true || !true) == false
FF 99: (!false || !true) == true
Chromium: (!false || !false) == true

In every case, `!Cypress.isBrowser('firefox')` has no effect on the outcome (because it's already part of the "is FF before 98" check).

Reply:
You're right. 

`isFirefoxBefore98` is `isFirefox && version < 98`.

`!isFirefoxBefore98 || !isFirefox` is `!isFirefox || !(version < 98) || !isFirefox`. 

I fixed it.